### DISCUSSION TITLE
In Natml 1.0.19 , GPU inference result is strange

### DISCUSSION THREAD

**Discussion by DeepFocuser** [natmlx/natml-unity#31] [Q&A] [answered]
In the environment of window10, unity 2021.3.11f, NatML 1.0.19 version, a problem occurred while making Unity app with the ONNX model(segmentation) I made myself. cpu inference results are printed well, but The results of the gpu inference are not working properly. Is Direct11 a problem?  Help me

💥Additionally, the same phenomenon occurs from 1.0.11 to 1.0.19 versions.

**Comment by olokobayusuf** [comment, marked as answer]
@DeepFocuser , some DirectML kernels seem to produce the wrong results. As a result, we have defaulted to using the CPU until [1.0.18](https://docs.natml.ai/unity/insiders/changelog#1.0.18). To be safe, set the `MLModelData.computeTarget` to `CPUOnly`.

**Reply by DeepFocuser** [reply]
Can we look forward to the next version(1.0.20)?

**Reply by olokobayusuf** [reply]
This depends on DirectML updates/drivers, so it isn't something that we can fix. As time goes on, we expect a lot of kernels to be fixed. But this is controlled by MSFT, so the safest option is to verify that a given model works properly on DirectML; and if it doesn't, switch back to the CPU.

**Reply by DeepFocuser** [reply]
Thank got it. 

Personally, I experimented the model(MobileNet) on Low spec Computer(Intel HD Integrated Graphics Card ) in Gpu option turned on. But, My unity app was not working.  Just like you said, I identified that Wrong operation depends on DirectMl Updates/Drivers.

Thank you for  your `hard fastest` reply
